components/ciphertool/src/main/java/org/wso2/ciphertool/CipherTool.java

```diff
@@ -53,9 +57,22 @@ public static void main(String[] args) {
         Cipher cipher = KeyStoreUtil.initializeCipher();
         if (System.getProperty(Constants.CONFIGURE) != null &&
             System.getProperty(Constants.CONFIGURE).equals(Constants.TRUE)) {
-            loadXpathValuesAndPasswordDetails();
-            secureVaultConfigTokens();
-            encryptCipherTextFile(cipher);
+            Map<String, String> secretMap = Utils.getSecreteFromConfiguration(Utils.getDeploymentFilePath());

```

Comment:
1. Check whether the toml file is available.
2. Check why you haven't call loadXpathValuesAndPasswordDetails, secureVaultConfigTokens, encryptCipherTextFile, if there are no secrets
3. Add tests to test this section